proc/proc.go

```diff
@@ -412,13 +412,39 @@ func (dbp *Process) Step() (err error) {
 
 // Change from current thread to the thread specified by `tid`.
 func (dbp *Process) SwitchThread(tid int) error {
+	fmt.Printf("SwitchThread: %d\n", tid)
+	if dbp.CurrentThread != nil && dbp.CurrentThread.Id == tid {
```

Comment:
Let's drop this branch and just let the map lookup below happen, and let `SelectedGoroutine` get set that way.